lib/experimental/enqueue-webfonts-listed-in-theme-json.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Emit inline CSS for allowed webfonts listed in theme.json.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Check if the webfont was registered programmatically.
+ *
+ * @param array $webfont The webfont that'll have their origin checked.
+ *
+ * @return boolean True if registered programmatically, false otherwise.
+ */
+function gutenberg_is_externally_registered_webfont( $webfont ) {
+	return isset( $webfont['origin'] ) && 'gutenberg_wp_webfonts_api' === $webfont['origin'];
+}
+
+/**
+ * Enqueue webfonts listed in theme.json.
+ *
+ * Enqueued webfonts will end up in the front-end as inlined CSS.
+ */
+function gutenberg_enqueue_webfonts_listed_in_theme_json() {
+	$theme_settings = WP_Theme_JSON_Resolver_Gutenberg::get_theme_data()->get_settings();
+
+	// Bail out early if there are no settings for webfonts.
+	if ( empty( $theme_settings['typography'] ) || empty( $theme_settings['typography']['fontFamilies'] ) ) {
+		return;
+	}
+
+	// Look for fontFamilies.
+	foreach ( $theme_settings['typography']['fontFamilies'] as $font_families ) {
```

Comment:
Can we use `get_merged_data()` instead of `get_theme_data()` here?
In https://github.com/WordPress/gutenberg/pull/39886/files we switched to that, because when a user selects a styles variation, the data defined there is saved as user-data. So any webfonts that are defined in variations will get picked-up when using `get_merged_data()` and get properly enqueued on the frontend.

```suggestion
	$settings = WP_Theme_JSON_Resolver_Gutenberg::get_merged_data()->get_settings();

	// Bail out early if there are no settings for webfonts.
	if ( empty( $settings['typography'] ) || empty( $settings['typography']['fontFamilies'] ) ) {
		return;
	}

	// Look for fontFamilies.
	foreach ( $settings['typography']['fontFamilies'] as $font_families ) {
```